safegraph_py_functions/safegraph_py_functions.py

```diff
@@ -225,3 +249,31 @@ def read_pattern_multi(path_to_pattern, compression='gzip', *args, **kwargs):
 def merge_core_pattern(core_df, patterns_df, how='inner', *args, **kwargs):
     merged_df = pd.merge(core_df, patterns_df, on='safegraph_place_id', how=how, *args, **kwargs)
     return merged_df
+
+
+### --------------------------------------- END CORE, GEO, AND PATTERNS SECTION -----------------------------------------------
+
+### --------------------------------------- SOCIAL DISTANCING SECTION -----------------------------------------------
+
+## start_range and end_range = string formated as "year-month-day" 
+    ## ex: start_range = "2020-06-01", end_range = "2020-06-07"
+
+def merge_socialDist_by_dates(path_to_social_dist, start_date, end_date, *args, **kwargs):
+    path = Path(path_to_social_dist,start_date[:4]) # go to year 
```

Comment:
@bpblakely what exactly is the `Path` module and function doing here? For the rest of the repo we require the user to pass correct and exact strings. I think it is OK if we require the user to do that here also, and then users don't have to import the `pathlib` library. All things equal, I would prefer to have as few extra libraries/dependencies as possible unless required. But I might be missing what value is being added with this function?

Reply:
Pathlib is just what library I use to easily go through paths, but everything I'm doing can be done with just importing os (I'll work on that code real quick).

path = Path(path_to_social_dist,start_date[:4]) is basically just changing the directory from: 'safegraph\social_distancing' to 'safegraph\social_distancing\2020' or 2019 if that is their start date.

The only thing that needs to be imported is 'calendar', since I use it for finding the end date of the month. I can probably find a way to do it with datetime if you want to, but it's way easier importing calendar.